Markdown output

mariusmuntean opened the pull request:
Added basic support to output the results as a markdown table.
Fixes #115

mariusmuntean committed:
feat: Add markdown output

Added simple markdown support. Outputs the results as a markdown table.

Fixed #115 (+1 squashed commit)
Squashed commits:
[63daad5] Added Markdown formatter

BePo65 commented on test/endToEnd.test.js:
```diff
@@ -344,3 +360,18 @@ const EXPECTED_TABLE_TEMPLATE = `{{department}}  {{relatedTo}}  {{name}}  {{lice
+	{{key}} - value to be replaced with value from package information
+	[[package-name]] - name of the package
+*/
+const EXPECTED_MARKDOWN_TABLE_TEMPLATE = 
```
I am missing the placeholders in the 'EXPECTED_MARKDOWN_TABLE_TEMPLATE' (comparable to the 'EXPECTED_TABLE_TEMPLATE'). This way, any test using this constant as an expected value will fail, as soon as a newer version is published in the npm repository.

mariusmuntean replied:
I missed that, but it should be fixed now.

BePo65 commented:
Sorry for the delay, but it's been only today that I finished a quite complex pr for the 'dayjs' package.

Regarding the configuration, I think we could simply add an entry to the `config.js` e.g. named `tablemarkOptions` with an empty literal object (`{}`) as a default value. If not empty, we could add this as a second parameter to the line 109 in `lib/getFormatter.js` ('return tablemark(dataAsArrayWithRenamedFields)').  
By default, does not require any configuration, as it uses the field names as column titles.

What do you think about this?

mariusmuntean commented:
I fully agree and I'll update the pr tomorrow or so.

mariusmuntean committed:
Allow passing custom tablemark config options.

mariusmuntean commented:
@BePo65 I've added the tablemark options passthrough. It wasn't necessary to check if the tablemark options object is empty.
I've also updated the README.md a bit.
What do you think?

BePo65 reviewed (commented):
Sorry, but I forgot to **submit** my review (silly error 😄 ).
Would you please add the placeholders in the template of the test results?

The rest is very well (even with good comments 👍 ).

BePo65 commented:
Thanks for adding the configuration section and especially of thinking of the README.md file.

mariusmuntean committed:
Fixed tests

mariusmuntean commented:
I saw the placeholders in the templates initially, but then completely forgot about them.

BePo65 reviewed (approved):
looks good to me
